job/job.go

```diff
@@ -232,11 +232,30 @@ func (sg *scopedGroup) Add(jobs ...Job) {
 	sg.group.add(sg.health, jobs...)
 }
 
-var nameRegex = regexp.MustCompile(`^[a-zA-Z][a-zA-Z0-9_\-]{0,100}$`)
-
-func validateName(name string) error {
-	if !nameRegex.MatchString(name) {
-		return fmt.Errorf("invalid job name: %q, expected to match %q", name, nameRegex)
+const maxNameLength = 100
```

Comment:
Is this grapheme clusters/runes or ascii chars?

Reply:
supposed to be ascii as we're looking at the length after mangling it to ascii. could you double-check the logic?